source/renderer/app/components/assets/AssetSettingsDialog.tsx

```diff
@@ -88,11 +88,11 @@ class AssetSettingsDialog extends Component<Props, State> {
     super(props);
     const { asset } = props;
     const { decimals: savedDecimals, recommendedDecimals } = asset;
-    const hasSavedDecimals = typeof savedDecimals === 'number';
+    const hasRecommendedDecimals = typeof recommendedDecimals === 'number';
     this.state = {
-      decimals: hasSavedDecimals
-        ? savedDecimals
-        : recommendedDecimals || DEFAULT_DECIMAL_PRECISION,
+      decimals: hasRecommendedDecimals
```

Comment:
The correct logic is: when the user opens the settings, if the token has recomended decimal, it should use it as the initial value, so it can be saved right away (in fact - fixed). If there is no recommendation we should show the current or fallback value.

Reply:
What if the user already has saved decimals? Would it display the recommended anyway?

Reply:
That is my understanding of the requirement. @dmitrii-gaico could you confirm?

Reply:
Yeah, that also seems a bit weird to me 🤔 I would also think that the saved decimals always trump the recommended ones (if there was anything saved ofc) like it was before

Reply:
Well, so I personally believe we should never change the initial form value, we should always show the current setting no matter what is recommended. As a user, when I open the edit form, I expect to see the currently applied values, no matter if these are defaults or set by me explicitly. Unless that value can no longer be used (eg. because the value selected is no longer available).

Initially when I tried to understand what was the problem, I really couldn't wrap my head around the UX of this form until I jumped on a call with @dmitrii-gaico and he explained it to me.

I would personally show a red screaming alert that the current setting is different from recommended setting (instead of tiny warning sign with no text) and give the user the chance to decide. Or at least add one more section that shows the current setting, because at the moment the user cannot clearly see what is the currently used decimal (or maybe I still don't understand this feature).

But - this is my personal opionion - and I guess this was already discussed in the past and carefully thought through.

Reply:
To be fair, as a user of multiple wallets, this is something unique to Daedalus and I was also initially confused in the beginning. But in my experience, initial state is initial (default) state unless there has been a write operation where that bit of state has been overridden.

Reply:
As discussed with @dmitrii-gaico, this change should be reverted.